src/expr/node.h

```diff
@@ -239,6 +239,22 @@ class NodeTemplate {
 
 public:
 
+  /**
+   * Cache-aware, recursive version of substituteCaptureAvoiding() used by the
+   * public member function with a similar signature.
+   */
+  Node substituteCaptureAvoiding(Node node, Node replacement,
```

Comment:
I would prefer having those functions in `node_algorithm` in the interest of compilation times since they are not very commonly used.